### DISCUSSION TITLE
increase/ deacrese indent and align dont display proprely / work

### DISCUSSION THREAD

**Discussion by CorentinFortes** [julianpoemp/ngx-jodit#30] [Q&A] [answered]
hi my some fonctionality dont work i would like to know if they are just not supported or if i can use them

here u can see the expected 
<img width="67" alt="Capture d'écran 2024-02-02 112858" src="https://github.com/julianpoemp/ngx-jodit/assets/91876233/fa4c9144-68e7-4329-9799-4909d83eef30">
and hre you can see what i have
<img width="102" alt="Capture d'écran 2024-02-02 113248" src="https://github.com/julianpoemp/ngx-jodit/assets/91876233/c29c301c-3f9b-4b6c-93fb-b17684082545">

btw here is my config 
```
editor: JoditConfig = {
    style: {
      fontFamily: 'Readex Pro',
      "white-space": 'pre-wrap'
    },
    readonly: false,
    minHeight: 100,
    maxHeight: 275,
    allowResizeX: false,
    allowResizeY: false,
    autofocus: false,
    language: "fr",
    enter: 'div',
    defaultMode: 1,
    toolbarAdaptive: false,
    // poweredByJodit: false,
    showCharsCounter: true,
    limitChars: 2000,
    limitHTML: false,
    showWordsCounter: false,
    showXPathInStatusbar: false,
    askBeforePasteHTML: false,
    askBeforePasteFromWord: false,
    hidePoweredByJodit: true,
    disablePlugins: 'add_new_line, enter',
    showPlaceholder: true,
    useInputsPlaceholder: true,
    placeholder: 'Libelle du composant',
    buttons: ['undo', 'redo','|','font','fontsize','brush','bold','italic','underline','superscript','subscript','|','align','indent','outdent','ul','ol'],
    controls: {
      fontsize:{
        list: {
          '8': '8',
          '9': '9',
          '10': '10',
          '11': '11',
          '12': '12',
          '14': '14',
          '16': '16',
          '18': '18',
          '24': '24',
        }
      },
      font: {
        list: {
          'Readex Pro': 'Readex Pro',
        }
      }
    },
    toolbarButtonSize: 'large',
    colorPickerDefaultTab: 'color',
    cleanHTML: {
      fillEmptyParagraph: false
    },
    i18n: {
      fr: {
        'Chars: %d': 'Caractères: %d',
        'Lower Alpha': 'Petit Alpha',
        'Upper Alpha': 'Grand Alpha',
        'Lower Roman': 'Petit chiffre romain',
        'Upper Roman': 'Grand chiffre romain',
        'Lower Greek': 'Lettre grecque',
      }
    },
    colors: {
      palette: [
        '#000001','#434343','#666666','#999999','#B7B7B7','#CCCCCC','#D9D9D9','#EFEFEF','#F3F3F3','#FFFFFF',
        '#980000', '#FF0000','#FF9900','#FFFF00','#00F0F0','#00FFFF','#4A86E8','#0000FF','#9900FF','#FF00FF',
        '#CC4125','#E06666','#F6B26B','#FFD966','#93C47D','#76A5AF','#6D9EEB','#6FA8DC','#8E7CC3','#C27BA0',
        '#A61C00','#CC0000','#E69138','#F1C232','#6AA84F','#45818E','#3C78D8','#3D85C6','#674EA7','#A64D79',
        '#85200C','#990000','#B45F06','#BF9000','#38761D','#134F5C','#1155CC','#0B5394','#351C75','#733554'
      ]
    }
  };
```

**Comment by julianpoemp** [comment]
Hi,
at first I thought it's an easy fix you should just import the proper plugins to allow text alignment (see [here](https://github.com/julianpoemp/ngx-jodit/blob/main/libs/ngx-jodit/README.md#how-to-import-plugins)). But it seems to be more difficult. I'll try to find a solution

**Reply by julianpoemp** [reply]
@CorentinFortes seems to be a bug on Jodit. I created an issue here: https://github.com/xdan/jodit/issues/1074. Please thumb up the issue to get solved faster.

**Reply by CorentinFortes** [reply]
hi I am really sorry to comme back this late but i wasn't on this subject so i totaly forgot to answer,
so here is my screen i actualy have
<img width="416" alt="image" src="https://github.com/julianpoemp/ngx-jodit/assets/91876233/72a6c99a-ea6f-4ced-b0a9-c07d4d24e0e6">
while i should have this 
<img width="399" alt="image" src="https://github.com/julianpoemp/ngx-jodit/assets/91876233/e27c7886-5cca-4cf6-b318-f2a22bdc890e">
as u can see i sill have no icons and no possibilities to set the alignement while some other buttons with the select button are available

I sill have the config of the previous for joddit.

thank you very much have a good day

**Reply by julianpoemp** [reply, marked as answer]
@CorentinFortes remember each button is a plugin itself. If you don't see an icon it indicates a missing plugin import. In your case import

```typescript
import 'jodit/esm/plugins/indent/indent.js';
import 'jodit/esm/plugins/justify/justify.js';
```

as described here: https://github.com/julianpoemp/ngx-jodit/blob/main/libs/ngx-jodit/README.md#how-to-import-plugins

I didn't know that you have to import "justify.js" to get the align dropdown button. Because there is no documentation you have to search the plugins from the node_modules folder and test some plugins.

**Comment by CorentinFortes** [comment]
Ty so much this was the probleme thank you very much again for your great reactivity and help
